cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/SlowBrokerFinder.java

```diff
@@ -200,15 +200,19 @@ private void collectLogFlushTimeMetric(BrokerEntity broker,
     AggregatedMetricValues aggregatedMetricValues = currentMetricsByBroker.get(broker).metricValues();
     double latestLogFlushTime = aggregatedMetricValues.valuesFor(BROKER_LOG_FLUSH_TIME_MS_999TH_ID).latest();
     currentLogFlushTimeMetricValues.put(broker, latestLogFlushTime);
-    aggregatedMetricValues = metricsHistoryByBroker.get(broker).metricValues();
-    double[] historicalLogFlushTime = aggregatedMetricValues.valuesFor(BROKER_LOG_FLUSH_TIME_MS_999TH_ID).doubleArray();
-    List<Double> historicalValue = new ArrayList<>(historicalLogFlushTime.length);
-    for (int i = 0; i < historicalLogFlushTime.length; i++) {
-      if (historicalLogFlushTime[i] > 5.0) {
-        historicalValue.add(historicalLogFlushTime[i]);
+    if (metricsHistoryByBroker.get(broker) != null) {
+      aggregatedMetricValues = metricsHistoryByBroker.get(broker).metricValues();
+      double[] historicalLogFlushTime = aggregatedMetricValues.valuesFor(BROKER_LOG_FLUSH_TIME_MS_999TH_ID).doubleArray();
+      List<Double> historicalValue = new ArrayList<>(historicalLogFlushTime.length);
+      for (int i = 0; i < historicalLogFlushTime.length; i++) {
+        if (historicalLogFlushTime[i] > 5.0) {
+          historicalValue.add(historicalLogFlushTime[i]);
+        }
       }
+      historicalLogFlushTimeMetricValues.put(broker, historicalValue);
+    } else {
+      LOG.info("Metric history for broker {} is missing.", broker.brokerId());
```

Comment:
(applies to the other log below) Is it possible to add reason regarding why it might be missing? Would a cold start for CC also cause this? Should this be an `info` or debug level log? -- i.e. would it be important for the user to see this in the log or is it internal to the slow broker finder design?